emily/handler/src/api/handlers/internal.rs

```diff
@@ -0,0 +1,220 @@
+                        new_reorg_tip,
+                        current_reorg_tip,
+                        ))));
+                }
+            }
+            (ApiStatus::Stable(new_tip), ApiStatus::Stable(old_tip)) => {
+                if new_tip == old_tip {
+                    return Ok(None);
+                } else {
+                    new_status.clone()
+                }
+            }
+            (ApiStatus::Stable(_), ApiStatus::Reorg(_)) => new_status.clone(),
+        };
+
+        debug!(
+            "Changing Api state from [{:?}] to [{:?}]. Attempt {} of maximum {}.",
+            original_api_state, api_state, update_attempts, MAX_SET_API_STATE_ATTEMPTS_DURING_REORG,
+        );
+
+        // Attempt to set the API state.
+        match accessors::set_api_state(context, &api_state).await {
+            // We successfully set the API state.
+            Ok(()) => {
+                info!("Successfully set api state: {:?}.", api_state);
+                break;
+            }
+            // Retry if there was a version conflict.
+            Err(Error::VersionConflict) => {
+                if update_attempts >= MAX_SET_API_STATE_ATTEMPTS_DURING_REORG {
+                    debug!("Failed to update API state {:?}", api_state);
+                    return Err(Error::InternalServer);
+                } else {
+                    debug!("Failed to update API state - retrying: {:?}", api_state);
+                }
+            }
+            // If some other error occured then return from here; this shouldn't
+            // happen and something has actually gone wrong.
+            Err(e) => Err(e)?,
+        }
+    }
+    // Return.
+    Ok(Some(api_state))
+}
+
+/// Handler that executes a reorg.
+///
+/// This function isn't intendeded to be exposed into any specific endpoint
+/// outside of what could maybe be a testing endpoint one day. It handles
+/// the internal requests to execute a reorg.
+pub async fn execute_reorg_handler(
+    context: &EmilyContext,
+    request: ExecuteReorgRequest,
+) -> Result<impl warp::reply::Reply, Error> {
+    info!("Executing a reorg with request {request:?}.");
+    let empty_reply = warp::reply::with_status(warp::reply(), StatusCode::NO_CONTENT);
+
+    let new_status = ApiStatus::Reorg(request.canonical_tip.clone().into());
+    match set_api_state_status(context, &new_status).await {
+        // Do nothing if we claimed the api correctly.
+        Ok(Some(_)) => {}
+        Ok(None) => {
+            return Ok(empty_reply);
+        }
+        Err(e) => {
+            return Err(e);
+        }
+    };
+
+    // We have control of the API at this point. For each entry of the deposit
+    // and withdrawal table we'll wipe out all the history that's no longer relevant.
+
+    // Get all deposits that would be impacted by this reorg.
+    let all_deposits = accessors::get_all_deposit_entries_modified_after_height(
```

Comment:
Yes. This will get all deposits / withdrawals that could have been impacted by the reorg.

Reply:
I suppose that means all requests at a height less than the input height?

Reply:
Less than or equal to, because the one given is being reorg so it too needs to be reorged

Reply:
Okay, I think I see. That the function name gave me the impression that we care about heights greater than the input height. Maybe we should rename it, or clarify that "after" here means that it happened before/less than the input height.

Reply:
I'll change it to just be after the height like you're suggesting and then put `height - 1` into the function.

Reply:
Wait, that's not quite what I was suggesting. The name `get_all_deposit_entries_modified_after_height` suggest to me that it will get all deposit entries that were "confirmed"/"handled" on a block height that was greater than (or greater than or equal to) the given input block height. But I think you just said that id does not do that, but instead it gets all deposit entries that happened at an earlier block height. That to me suggests that the function is misnamed (or that I misunderstand something). So you don't need to change the usage here.